lib/mkl/linalg.jl

```diff
@@ -49,6 +49,13 @@ function gemm_dispatch!(C::oneStridedVecOrMat, A, B, alpha::Number=true, beta::N
     end
 end
 
+LinearAlgebra.rmul!(x::oneStridedVecOrMat{<:onemklFloat}, k::Number) = 
```

Comment:
I am not sure if these two rules will properly dispatch the combination of alpha types of ComplexF32, ComplexF64, F32 and F64 to the ComplexF32 and Complex64 scal functions.

Please write specific tests for all the combinations to make sure they are dispatched properly.

If not, we will want to use AMDGPU.jl's type based dispatching rules at
 https://github.com/JuliaGPU/AMDGPU.jl/blob/master/src/blas/wrappers.jl#L85, and

https://github.com/JuliaGPU/AMDGPU.jl/blob/master/src/blas/wrappers.jl#L106

Reply:
Sure, Thanks